### PR TITLE
More error handling

### PR DESCRIPTION
**Pull request by matsumotory**
ref: https://github.com/haconiwa/mruby-mount/pull/11

- Add sys_fail wrapper which support va_args and errno message
- Simplify code

@udzura Please review it.

### CODE REVIEW

**Comment by matsumotory** [comment]
@udzura I simplify sys error message by removing file and function name. Please check it.

File: test/mrb_exec.rb [hunk 1]
```diff
@@ -28,3 +28,14 @@
   assert_true(p.is_a?(Integer))
   assert_true(status.success?)
 end
+
+assert("Kernel#exec sys fail") do
+  expected_error_message = "sys failed. errno: 2 message: No such file or directory mrbgem message: execv failed"
+
+  begin
+    # always failed
+    p = Process.fork { exec "/bin/bbaash" }
```
Comment (udzura) [inline test/mrb_exec.rb]: This kind of `exec` will always fail, so `fork` would not be required?

Comment (udzura) [reply test/mrb_exec.rb]: Will this pass?

In this patch if L39 is ignored, the test will be marked as `passed` anyway.

```ruby
e = nil
begin
  _p = Process.fork { exec "/bin/bbaash" }
rescue => _e
  e = _e
end
assert_false e.nil?
assert_equal(expected_error_message, e.to_s)
```

**Review by udzura — CHANGES_REQUESTED** [review]

File: test/mrb_exec.rb (hunk 1 above)
Comment (matsumotory) [reply test/mrb_exec.rb]: Thank you for your review. I fixed via https://github.com/haconiwa/mruby-exec/pull/8/commits/b3cae2cb7e3c8ff07b0b428ab2fc471dc7dd3553

Comment (udzura) [reply test/mrb_exec.rb]: b3cae2c is also OK!!

**Review by udzura — APPROVED** [review]

**Pull request merged**